### PR TITLE
Specific response assertion error messages

### PR DESCRIPTION
**Pull request by PietervdWerk**
I've been getting the error `"failed to get response body: type assertion error"` and it took me quite a while to figure out what the problem was. I've copied my policy into a playground and it worked fine there (in hindsight that's obvious).

I propose a more verbose and specific error message when assertion fails. I've taken the liberty to come with some suggestions.

### CODE REVIEW

File: envoyauth/response.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func (result *EvalResult) GetDynamicMetadata() (*_structpb.Struct, error) {
 
 		metadata, ok := val.(map[string]interface{})
 		if !ok {
-			return nil, fmt.Errorf("type assertion error")
+			return nil, fmt.Errorf("type assertion error, dynamic_metadata is not a map[string]interface{}")
```
Comment (anderseknert) [inline envoyauth/response.go:310]: `map[string]interface{}` is a Go type, not a Rego one. I'd just say "is not an object" here. Same for headers below.

**Review by anderseknert — COMMENTED** [review]
Thanks! Looks good to me, with a minor change suggested. I wonder if it would be helpful to go one step further and say "expected x, got y" in these messages? But this is an improvement nonetheless.

**Review by anderseknert — DISMISSED** [review]
LGTM! @ashutosh-narkar ?

**Review by ashutosh-narkar — COMMENTED** [review]
@PietervdWerk the changes lgtm. It would be great if we can update the tests to reflect these detailed error messages.

**Comment by ashutosh-narkar** [comment]
@PietervdWerk this would be a good change to get in. Are you able to update the tests?

**Comment by PietervdWerk** [comment]
Yes tomorrow! (I had a few days off, back to work tomorrow.)

**Review by ashutosh-narkar — APPROVED** [review]
Thanks @PietervdWerk!

**Pull request merged**